### ISSUE TITLE
Planning / calendar not working with macOS calendar

### ISSUE THREAD

**Issue by blippercop** [glpi-project/glpi#12990]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.3

### Bug description

I am unable to subscribe to the CalDAV calendar on macOS.
Nothing in the user docs that macos is not supported.

### Relevant log output

```shell
[2022-10-15 22:08:20] glpiphplog.CRITICAL:   *** Uncaught Exception Sabre\DAV\Exception\NotImplemented: There was no plugin in the system that was willing to handle this GET method. Enable the Browser plugin to get a better result here. in /var/www/helpdesk/vendor/sabre/dav/lib/DAV/Server.php at line 479
  Backtrace :
  vendor/sabre/dav/lib/DAV/Server.php:253            Sabre\DAV\Server->invokeMethod()
  caldav.php:50                                      Sabre\DAV\Server->start()
```

### Steps To reproduce

1. go to front/planning.php
2. copy your Caldav URL
3. use Calendar application on mac -> subscribe to calendar
4. type in your GLPI credentials
5. see error that the calendar cannot be added -> "unexpected error" 
6. see error in GLPI logs

**Comment by cedric-anne** [comment]
> What I've noticed is that when the GLPI calendar is added to the macOS application the GLPI protocol (/front/event.php) is literally flooded(!) with "USER" login from IP XXX.XXX.XXX.XXX on every update (request) the calendar does with a hundred/thousand entries - which make the log unreadable as it is only showing the calendar updates excessively (and I only added one user, yet). Is this normal / by design?

It is not normal that it triggers access to `/front/event.php`, only `caldav.php` endpoint should be accessed. However, it is possible that the macOS application you use is doing a query to GLPI for each event that is served by the CalDAV server.

Can you try to apply following patch to see if it prevents requests on `/front/event.php` ?
```diff
diff --git a/src/CalDAV/Traits/VobjectConverterTrait.php b/src/CalDAV/Traits/VobjectConverterTrait.php
index 40171ee5d7..8f42675742 100644
--- a/src/CalDAV/Traits/VobjectConverterTrait.php
+++ b/src/CalDAV/Traits/VobjectConverterTrait.php
@@ -131,7 +131,7 @@ trait VobjectConverterTrait
             $vcomp->DESCRIPTION = RichText::getTextFromHtml($description, true);
         }
 
-        $vcomp->URL = $CFG_GLPI['url_base'] . $this->getFormURLWithID($fields['id'], false);
+        //$vcomp->URL = $CFG_GLPI['url_base'] . $this->getFormURLWithID($fields['id'], false);
 
         if (array_key_exists('begin', $fields) && !empty($fields['begin'])) {
             $vcomp->DTSTART = (new \DateTime($fields['begin']))->setTimeZone($utc_tz);
diff --git a/src/CommonITILTask.php b/src/CommonITILTask.php
index ec9ba517d8..9dbaf287cc 100644
--- a/src/CommonITILTask.php
+++ b/src/CommonITILTask.php
@@ -1804,7 +1804,7 @@ abstract class CommonITILTask extends CommonDBTM implements CalDAVCompatibleItem
         $vcomp->SUMMARY           = $parent_fields['name'];
         $vcomp->DTSTAMP           = (new \DateTime($parent_fields['date_mod']))->setTimeZone($utc_tz);
         $vcomp->{'LAST-MODIFIED'} = (new \DateTime($parent_fields['date_mod']))->setTimeZone($utc_tz);
-        $vcomp->URL               = $CFG_GLPI['url_base'] . $parent_item->getFormURLWithID($parent_id, false);
+        //$vcomp->URL               = $CFG_GLPI['url_base'] . $parent_item->getFormURLWithID($parent_id, false);
 
         return $vcalendar;
     }
diff --git a/src/ProjectTask.php b/src/ProjectTask.php
index 2bc32b5432..b4c5d4f684 100644
--- a/src/ProjectTask.php
+++ b/src/ProjectTask.php
@@ -2031,7 +2031,7 @@ class ProjectTask extends CommonDBChild implements CalDAVCompatibleItemInterface
             }
         }
 
-        $vcomp->URL = $CFG_GLPI['url_base'] . Project::getFormURLWithID($fields['projects_id'], false);
+        //$vcomp->URL = $CFG_GLPI['url_base'] . Project::getFormURLWithID($fields['projects_id'], false);
 
         return $vcalendar;
     }
```

**Comment by patshining** [comment]
I've got the same problem with Outlook and Caldav Synchronizer since GLPI v10.0.x (v10.0.10 now). 

I'm using Apache2 configured like recommended with "DocumentRoot /path/glpi/public" + directory "<Directory /var/www/html/glpi/public>" and the rewrite cond.

I've got a 403 error on Caldav Synchronizer and this line on the access apache2 log file : 
"PROPFIND /caldav.php/calendars/users/J.DUPONT/calendar/ HTTP/1.1" 403 2017 "-" "CalDavSynchronizer/4.4"

If i remove "public" on the DocumentRoot, Caldav Synchronizer works.
